### DISCUSSION TITLE
Import previous data

### DISCUSSION THREAD

**Discussion by arthvpatel** [teslamate-org/teslamate#4366] [Ideas]
I know the streaming api does not provide historical data but you can export your history, I just did and it provides how much charging it did where and when, it also had a bunch of information

How to Request a Copy of Your Data
Step 1
Go to www.tesla.com/en_ca/contactus and sign in to your Tesla Account.
Step 2
Use the drop-down menu to select 'Data Privacy Request.' Specify the time range for which you would like to request your data. Select 'Submit.'
Step 3
We will verify your request and notify you by email when your data is ready. As a reminder, this process can take up to 30 days.

**Comment by NorthStar7** [comment]
Thanks for the heads up. I just submitted a request.  My TeslaMate cloud VM disk is full and I cannot figure out how to get my data out of it, or otherwise get it back up and running.  If the Tesla data is sufficient, maybe I can abandon my $5/month paperweight in the cloud.

**Reply by arthvpatel** [reply]
> Thanks for the heads up. I just submitted a request.  My TeslaMate cloud VM disk is full and I cannot figure out how to get my data out of it, or otherwise get it back up and running.  If the Tesla data is sufficient, maybe I can abandon my $5/month paperweight in the cloud.

I run mine on google cloud using the free tier, it is a bit slower than locally hosting it but good enough for me to have it run 24/7. I end up paying 20cents a month for bandwidth over usage

**Reply by NorthStar7** [reply]
Thanks. I'll check out Google Cloud's offering.  What machine type do you use?

I actually did get my TeslaMate back up and running last night...the Docker JSON log file grew to over 14GB, filled my quota, and basically made it unusable.  I am an on-prem Windows/SQL Server guy so troubleshooting Cloud, Ubuntu, Docker, PostgreSQL, etc. was a learning experience.  I prefer to avoid local hosting as to not increase the surface area of my network accessible to the internet.  If I was a security guru and knew what I was doing, I am sure I could do it confidently, but I am not...

**Reply by arthvpatel** [reply]
> Thanks. I'll check out Google Cloud's offering.  What machine type do you use?
> 
> 
> 
> I actually did get my TeslaMate back up and running last night...the Docker JSON log file grew to over 14GB, filled my quota, and basically made it unusable.  I am an on-prem Windows/SQL Server guy so troubleshooting Cloud, Ubuntu, Docker, PostgreSQL, etc. was a learning experience.  I prefer to avoid local hosting as to not increase the surface area of my network accessible to the internet.  If I was a security guru and knew what I was doing, I am sure I could do it confidently, but I am not...

I don't expose anything to the web, just use Tailscale to connect to it, E2 micro, 2 core 1gb ram, 30gb hdd. follow the first link for the setup including swap, 2nd link for the current free tiers 

https://www.teslaev.co.uk/how-to-setup-and-run-teslamate-for-free-on-google-cloud/

https://cloud.google.com/free/docs/compute-getting-started#:~:text=Each%20month%2C%20eligible%20use%20of,for%20an%20external%20IP%20address.
